main/src/main/java/com/sedmelluq/discord/lavaplayer/track/playback/LocalAudioTrackExecutor.java

```diff
@@ -99,44 +103,75 @@ public void execute(TrackStateListener listener) {
       log.debug("Starting to play track {} locally with listener {}", audioTrack.getInfo().identifier, listener);
 
       state.set(AudioTrackState.LOADING);
+      attemptProcess(listener);
 
-      try {
-        audioTrack.process(this);
+    } else {
+      log.warn("Tried to start an already playing track {}", audioTrack.getIdentifier());
+    }
+  }
 
-        log.debug("Playing track {} finished or was stopped.", audioTrack.getIdentifier());
-      } catch (Throwable e) {
-        // Temporarily clear the interrupted status so it would not disrupt listener methods.
-        interrupt = findInterrupt(e);
+  private void attemptProcess(TrackStateListener listener) {
 
-        if (interrupt != null && checkStopped()) {
-          log.debug("Track {} was interrupted outside of execution loop.", audioTrack.getIdentifier());
-        } else {
-          frameBuffer.setTerminateOnEmpty();
+    InterruptedException interrupt = null;
 
-          FriendlyException exception = ExceptionTools.wrapUnfriendlyExceptions("Something broke when playing the track.", FAULT, e);
-          ExceptionTools.log(log, exception, "playback of " + audioTrack.getIdentifier());
+    try {
+      audioTrack.process(this);
 
-          trackException = exception;
-          listener.onTrackException(audioTrack, exception);
+      log.debug("Playing track {} finished or was stopped.", audioTrack.getIdentifier());
+    } catch (Throwable e) {
 
-          ExceptionTools.rethrowErrors(e);
-        }
-      } finally {
-        synchronized (actionSynchronizer) {
-          interrupt = interrupt != null ? interrupt : findInterrupt(null);
+      // Check for 403, attempt to clear cipher cache and retry if no retries in the past 5 seconds.
+      if (e.getMessage().contains("Not success status code: 403")
+          && (lastRetry == -1 || lastRetry + RETRY_COOLDOWN <= System.currentTimeMillis())
+          && audioTrack.getSourceManager() instanceof YoutubeAudioSourceManager) {
+        lastRetry = System.currentTimeMillis();
 
-          playingThread.compareAndSet(Thread.currentThread(), null);
+        log.debug("Detected 403, clearing cipher cache and retrying.");
 
-          markerTracker.trigger(ENDED);
-          state.set(AudioTrackState.FINISHED);
-        }
+        YoutubeAudioSourceManager sourceManager = (YoutubeAudioSourceManager) audioTrack.getSourceManager();
+        DefaultYoutubeTrackDetailsLoader trackDetailsLoader = (DefaultYoutubeTrackDetailsLoader) sourceManager.getTrackDetailsLoader();
```

Comment:
https://github.com/sedmelluq/lavaplayer/blob/84f0d36a6b32a40bf9ab290ca5590d578ddc5d24/main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/YoutubeAudioSourceManager.java#L78-L86

Using `YoutubeSourceManager` without `DefaultYoutubeTrackDetailsLoader` throws `ClassCastException` at this line.

Reply:
Should be fixed in [29eecb1](https://github.com/sedmelluq/lavaplayer/pull/631/commits/29eecb1387be8327855cab201397270b62d1bb81)